langchain/retrievers/prompt_engineer.py

```diff
@@ -0,0 +1,53 @@
+from langchain.chat_models import ChatOpenAI
+from langchain.prompts import PromptTemplate
+
+class PromptEngineerRetriever:
+
+    """ Use an LLM to produce a specified number of altnerative prompts for retrival. Query the retriever with all prompts and return the unique results."""
+    
+    # Prompt 
+    QUERY_PROMPT = PromptTemplate(
+        input_variables=["question", "num_queries"],
+        template="""You are helping a user perform retrieval of documents from a vector database. 
+
+        Take the user question, and create {num_queries} queries that capture that modify the wording in order to maximize the likelihood that relevant documents are retrieved.
+
+        Query: {question}
+
+        Alterative queries with each seperated by a newline:"""
+    )
+
+    def __init__(self, retriever, num_queries):
+        """ Simply pass a retriever and a specified number of queries """
+        self.retriever = retriever
+        self.num_queries = num_queries
+        self.llm = self.init_llm()
+
+    def init_llm(self):
```

Comment:
this should be passed in

we should probably pass in an LLMChain, so user can fully customize model and prompt

but we can have a `from_llm` method that sets the default prompt (and maybe default llm but usually we make them pass in an LLM so they're aware its using one)